c_horiz_interp/c_horiz_interp.F90

```diff
@@ -108,6 +127,139 @@ subroutine cFMS_set_current_interp(interp_id) bind(C, name="cFMS_set_current_int
     
   end subroutine cFMS_set_current_interp
 
+  !cFMS_horiz_interp_dealloc
+  subroutine cFMS_horiz_interp_dealloc() bind(C, name="cFMS_horiz_interp_dealloc")
+    implicit none
+
+    if(allocated(interp)) deallocate(interp)
+    nullify(current_interp)
+
+  end subroutine cFMS_horiz_interp_dealloc
+
+  subroutine cFMS_get_i_src(interp_id, i_src) bind(C, name="cFMS_get_i_src")
+      integer, intent(in) :: interp_id
+      type(c_ptr), intent(in), value :: i_src
+      integer, pointer :: i_src_cf(:)
+      call cFMS_set_current_interp(interp_id)
+      if(C_ASSOCIATED(i_src)) then
```

Comment:
If I'm following correctly, i_src is malloc-ed before this subroutine call?  Since it's 1d, it's probably easier to avoid pointer action so that, `integer :: i_src(some_size)`.  And the same for the rest of the get* methods

Reply:
yeah that definitely makes more sense, added that in